### PR TITLE
bypass bogus AOM crashtests that assume abandoned syntax will be added.

### PR DESCRIPTION
**Pull request by cookiecrook**
Closes https://github.com/web-platform-tests/interop-2023-accessibility-testing/issues/56

### CODE REVIEW

**Review by gsnedders — COMMENTED** [review]
In broad strokes: I wonder if we should move all the AOM tests to a top-level `aom` directory, rather than having a miscellaneous `accessibility` directory—everywhere else we group things by spec, after all. This would also arguably make it easier to ignore tests for the abandoned spec, treating it the same as any other spec (say, WebUSB) which doesn't have cross-vendor support.

File: accessibility/crashtests/aom-in-destroyed-iframe.html [hunk 1]
```diff
@@ -1,6 +1,11 @@
 <html class="test-wait">
 <body></body>
 <script>
+
+// Bypass this abandoned API in all but the engines that implement it.
```
[the remark this answers is not in the document]
Comment (cookiecrook) [reply accessibility/crashtests/aom-in-destroyed-iframe.html:5]: CG is not considering that form... I'll update the explainer to mention the concept is not abandoned, but remove references to specific syntax.

File: accessibility/crashtests/aom-in-destroyed-iframe.html [hunk 2]
```diff
@@ -1,6 +1,11 @@
...
+if (typeof getComputedAccessibleNode !== 'undefined') {
+
+  // presumed chromium only
```
[the remark this answers is not in the document]
Comment (cookiecrook) [reply accessibility/crashtests/aom-in-destroyed-iframe.html:8]:
```suggestion
```

File: accessibility/crashtests/computed-accessible-text-node.html [hunk 3]
```diff
@@ -1,14 +1,25 @@
...
-  let img = document.getElementById('img1');
-  let computed_accessible_node = await self.getComputedAccessibleNode(img);
-  // The next sibling is a text node.
-  let next_sibling = computed_accessible_node.nextSibling;
-  await next_sibling.ensureUpToDate();
+
+// Bypass this abandoned API in all but the engines that implement it.
+if (typeof getComputedAccessibleNode !== 'undefined') {
+
+  // presumed chromium only
```
Comment (cookiecrook) [inline accessibility/crashtests/computed-accessible-text-node.html:8]:
```suggestion
```

File: accessibility/crashtests/computed-node-checked.html [hunk 4]
```diff
@@ -19,6 +19,11 @@
 </script>
 <script>step_timeout(gc, 50);</script>
 <script>
+
+// Bypass this abandoned API in all but the engines that implement it.
+if (typeof getComputedAccessibleNode !== 'undefined') {
+
+  // presumed chromium only
```
Comment (cookiecrook) [inline accessibility/crashtests/computed-node-checked.html:26]:
```suggestion
```

File: accessibility/crashtests/computed-accessible-child-of-pseudo-element.html [hunk 5]
```diff
@@ -6,17 +6,28 @@
...
-  // The first child of the heading is a pseudo element <table>.
-  await first_child.ensureUpToDate();
-  // The next child down has an accessibility object but no node.
-  let grand_child = first_child.firstChild;
-  await grand_child.ensureUpToDate();
+
+// Bypass this abandoned API in all but the engines that implement it.
+if (typeof getComputedAccessibleNode !== 'undefined') {
+
+  // presumed chromium only
```
Comment (cookiecrook) [inline accessibility/crashtests/computed-accessible-child-of-pseudo-element.html:13]:
```suggestion
```

File: accessibility/crashtests/aom-in-destroyed-iframe.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (cookiecrook) [reply accessibility/crashtests/aom-in-destroyed-iframe.html:5]: See a little lower in that same document:

> ##### Legacy and abondoned implementations
> A purely experimental implementation exists in Blink,
> via the command-line flag `--enable-blink-features="AccessibilityObjectModel"`.
> 
> This adds a method to `Window`, `getComputedAccessibleNode(node)`,

[the remark this answers is not in the document]
Comment (cookiecrook) [reply accessibility/crashtests/aom-in-destroyed-iframe.html:5]: Updated AOM explainer and related issue.

**Comment by cookiecrook** [comment]
> In broad strokes: I wonder if we should move all the AOM tests to a top-level `aom` directory, rather than having a miscellaneous `accessibility` directory—everywhere else we group things by spec, after all. This would also arguably make it easier to ignore tests for the abandoned spec, treating it the same as any other spec (say, WebUSB) which doesn't have cross-vendor support.

That directory is several years old, and I have no issue with moving it... However, it's only crashtests... I added a [ReadMe](https://github.com/web-platform-tests/wpt/blob/master/accessibility/ReadMe.md) back in the Spring that effectively says... "Don't put new tests here unless they fit nowhere else."

cc @mfreed7 and @aleventhal since they have committed to the dir recently.

**Comment by cookiecrook** [comment]
@gsnedders all the new accessibility tests are in spec-relevant directories: 
- https://wpt.fyi/results/?label=master&label=experimental&aligned&q=label%3Aaccessibility

File: accessibility/crashtests/aom-in-destroyed-iframe.html (hunk 1 above)
[the remark this answers is not in the document]
Comment (cookiecrook) [reply accessibility/crashtests/aom-in-destroyed-iframe.html:5]: I'll change "abandoned API" to "abandoned syntax", which I realize now may be what you were suggesting.

[the remark this answers is not in the document]
Comment (cookiecrook) [reply accessibility/crashtests/aom-in-destroyed-iframe.html:5]:
```suggestion
// Bypass this abandoned syntax in all but the engines that implement it.
```

File: accessibility/crashtests/computed-accessible-child-of-pseudo-element.html [hunk 6]
```diff
@@ -6,17 +6,27 @@
...
-  let heading = document.querySelector('h1');
-  let computed_accessible_node = await self.getComputedAccessibleNode(heading);
-  let first_child = computed_accessible_node.firstChild;
-  // The first child of the heading is a pseudo element <table>.
-  await first_child.ensureUpToDate();
-  // The next child down has an accessibility object but no node.
-  let grand_child = first_child.firstChild;
-  await grand_child.ensureUpToDate();
+
+// Bypass this abandoned API in all but the engines that implement it.
```
Comment (cookiecrook) [inline accessibility/crashtests/computed-accessible-child-of-pseudo-element.html:10]:
```suggestion
// Bypass this abandoned syntax in all but the engines that implement it.
```

File: accessibility/crashtests/computed-accessible-text-node.html [hunk 7]
```diff
@@ -1,14 +1,24 @@
...
 <img id="img1">text
 <script>
-async function trigger1() {
-  let img = document.getElementById('img1');
-  let computed_accessible_node = await self.getComputedAccessibleNode(img);
-  // The next sibling is a text node.
-  let next_sibling = computed_accessible_node.nextSibling;
-  await next_sibling.ensureUpToDate();
+
+// Bypass this abandoned API in all but the engines that implement it.
```
Comment (cookiecrook) [inline accessibility/crashtests/computed-accessible-text-node.html:5]:
```suggestion
// Bypass this abandoned syntax in all but the engines that implement it.
```

File: accessibility/crashtests/computed-node-checked.html [hunk 8]
```diff
@@ -19,6 +19,10 @@
 </script>
 <script>step_timeout(gc, 50);</script>
 <script>
+
+// Bypass this abandoned API in all but the engines that implement it.
```
Comment (cookiecrook) [inline accessibility/crashtests/computed-node-checked.html:23]:
```suggestion
// Bypass this abandoned syntax in all but the engines that implement it.
```

File: accessibility/crashtests/aom-in-destroyed-iframe.html [hunk 9]
```diff
@@ -1,6 +1,10 @@
 <html class="test-wait">
 <body></body>
 <script>
+
+// Bypass this abandoned API in all but the engines that implement it.
```
Comment (cookiecrook) [inline accessibility/crashtests/aom-in-destroyed-iframe.html:5]:
```suggestion
// Bypass this abandoned syntax in all but the engines that implement it.
```

**Comment by mfreed7** [comment]
> cc @mfreed7 and @aleventhal since they have committed to the dir recently.

I'll definitely defer to @aleventhal - I've only committed here once (happened to be a week ago) but that was just to correct an existing test.

**Comment by aleventhal** [comment]
LGTM on general approach -- I didn't review line by line.

**Review by spectranaut — APPROVED** [review]
looks fine to me!

File: accessibility/crashtests/aom-in-destroyed-iframe.html [hunk 10]
```diff
@@ -1,6 +1,11 @@
 <html class="test-wait">
```
[the remark this answers is not in the document]
Comment (limck620) [reply accessibility/crashtests/aom-in-destroyed-iframe.html:1]: []()

**Pull request merged**